src/git.ts

```diff
@@ -36,7 +42,33 @@ export class Git {
       });
   }
 
-  public static async remoteSha(repo: string, ref: string): Promise<string> {
+  public static async remoteSha(repo: string, ref: string, token?: string): Promise<string> {
+    const repoMatch = repo.match(/github.com\/([^/]+)\/([^/]+?)(?:\.git)?(\/|$)/);
+    // if we have a token and this is a GitHub repo we can use the GitHub API
+    if (token && repoMatch) {
+      core.setSecret(token);
+      const octokit = new (Octokit.plugin(restEndpointMethods).defaults({
+        baseUrl: GitHub.apiURL
+      }))({auth: token});
+      const [owner, repoName] = repoMatch.slice(1, 3);
+      return await octokit.rest.repos
```

Comment:
nit: it's a little odd to mix the `await` and `then` promise styles...i think this would be simpler as:

```
try {
  let resp = await ..listCommits()
  return resp.data[0].sha
} catch (e) {
  throw new Error(`Cannot find remote ref for ${repo}#${ref}: ${err.message}`);
}
```